src/UnigramWordPredictor.java

```diff
@@ -51,8 +52,25 @@ public UnigramWordPredictor(Tokenizer tokenizer) {
   public void train(Scanner scanner) {
     List<String> trainingWords = tokenizer.tokenize(scanner);
 
-    // TODO: Convert the trainingWords into neighborMap here
-  }
+    // Converts the trainingWords into neighborMap here
+      neighborMap = new HashMap<>();
+
+      for(int i = 0; i < trainingWords.size()-1; i++){
+        String key = trainingWords.get(i);
+        String next = trainingWords.get(i+1);
+        List<String> value = new ArrayList<>();
+        if(neighborMap.containsKey(key)){
```

Comment:
Nice! As an alternative, can you think about how you could write this using `putIfAbsent` or `computeIfAbsent` instead?